### PR TITLE
Change MD5 to SHA256 algorithm

### PR DESCRIPTION
**Pull request by zbud-msft**
#### Why I did it

Semgrep complains about use of MD5 and how it not secure, so will use sha256 sum 

#### How I did it

Use SHA256 hash instead of MD5

#### How to verify it

Manual testing

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: telemetry/telemetry.go [hunk 1]
```diff
@@ -112,11 +114,10 @@ func main() {
 			}
 			certificate, err = tls.LoadX509KeyPair(*serverCert, *serverKey)
 			if err != nil {
-				currentTime := time.Now().UTC()
-				log.Infof("Server Cert md5 checksum: %x at time %s", md5.Sum([]byte(*serverCert)), currentTime.String())
-				log.Infof("Server Key md5 checksum: %x at time %s", md5.Sum([]byte(*serverKey)), currentTime.String())
 				log.Exitf("could not load server key pair: %s", err)
 			}
+			validateSHA256(*serverCert)
+			validateSHA256(*serverKey)
```
Comment (ganglyu) [inline telemetry/telemetry.go:120]: Original implementation is: when tls.LoadX509KeyPair has error, we generate log with md5 checksum of file name.
New implementation is: when tls.LoadX509KeyPair has no error, we generate log with sha256 checksum of file content.
Can you explain why?

File: telemetry/telemetry.go [hunk 2]
```diff
@@ -209,3 +210,19 @@ func getflag(name string) string {
 	})
 	return val
 }
+
+func validateSHA256(file string) {
```
Comment (FengPan-Frank) [inline telemetry/telemetry.go:214]: Seems this function is returning void, does main function need to break if validation is failed?

**Comment by zbud-msft** [comment]
Opened new PR as now will use SHA512 checksum instead of 256 https://github.com/sonic-net/sonic-gnmi/pull/186

File: telemetry/telemetry.go (hunk 1 above)
Comment (zbud-msft) [reply telemetry/telemetry.go:120]: Should be generating log with sha512 checksum when loadx509 fails. Fixed in https://github.com/sonic-net/sonic-gnmi/pull/186

File: telemetry/telemetry.go (hunk 2 above)
Comment (zbud-msft) [reply telemetry/telemetry.go:214]: Called in block that will exit in new PR: https://github.com/sonic-net/sonic-gnmi/pull/186

**Pull request closed without merging**